roles/ceph-osd/templates/ceph-osd.service.j2

```diff
@@ -1,15 +1,25 @@
 # {{ ansible_managed }}
 [Unit]
 Description=Ceph OSD
+{% if container_binary == 'docker' %}
 After=docker.service
 Requires=docker.service
+{% else %}
+After=network.target
+{% endif %}
+{% set cpu_limit = ansible_processor_vcpus|int if ceph_osd_docker_cpu_limit|int > ansible_processor_vcpus|int else ceph_osd_docker_cpu_limit|int %}
 
 [Service]
 EnvironmentFile=-/etc/environment
-ExecStartPre=-/usr/bin/docker stop ceph-osd-%i
-ExecStartPre=-/usr/bin/docker rm -f ceph-osd-%i
+{% if container_binary == 'podman' %}
+ExecStartPre=-/usr/bin/rm -f /%t/%n-pid /%t/%n-cid
+{% else %}
```

Comment:
one of the requirement for using `pidfile` and `forking` for the osd template was to move everything from `ceph-osd-run.sh` to the systemd template because you can't managed the `%t` and `%n` values in the ceph-osd-run.sh script.